homeassistant/components/discovery.py

```diff
@@ -84,7 +84,8 @@
 CONFIG_SCHEMA = vol.Schema({
     vol.Required(DOMAIN): vol.Schema({
         vol.Optional(CONF_IGNORE, default=[]):
-            vol.All(cv.ensure_list, [vol.In(SERVICE_HANDLERS)])
+            vol.All(cv.ensure_list, [
+                vol.In({**CONFIG_ENTRY_HANDLERS, **SERVICE_HANDLERS})])
```

Comment:
This is inefficient, as you're creating a dictionary just to iterate over the values. More efficient would be `list(CONFIG_ENTRY_HANDLERS) + list(SERVICE_HANDLERS)`